### PR TITLE
phone input fixes

### PR DESCRIPTION
**Pull request by ovkulkarni**
2b37b34 None was causing an invalid phone number but it is the default value

### CODE REVIEW

File: intranet/apps/preferences/fields.py [hunk 1]
```diff
@@ -25,7 +22,5 @@ def to_python(self, value):
         return value
 
     def widget_attrs(self, widget):
-        attrs = {}
-        if self.max_length is not None:
-            attrs.update({'maxlength': str(self.max_length)})
+        attrs = {"maxlength": "17"}
```
Comment (pefoley2) [inline intranet/apps/preferences/fields.py]: please have a comment to explain this "magic value".

**Comment by pefoley2** [comment]
Don't you need to run makemigrations?

File: intranet/apps/preferences/fields.py (hunk 1 above)
Comment (pefoley2) [inline intranet/apps/preferences/fields.py]: even better would be a define that is used in all the occurrences.

Comment (ovkulkarni) [inline intranet/apps/preferences/fields.py]: @ezwang

**Comment by pefoley2** [comment]
Test please?

File: intranet/apps/preferences/fields.py (hunk 1 above)
Comment (ezwang) [inline intranet/apps/preferences/fields.py]: The maximum the regex will accept is 17, I'll add a commit in later.
We should standardize the phone fields and make them all display either "None" or be blank when they are empty. Currently, some of them display "None" when the field is empty and others are blank. I personally think that making them all blank is the best option.

Comment (pefoley2) [inline intranet/apps/preferences/fields.py]: It's possible the None is left over from old db entries?

Comment (ovkulkarni) [inline intranet/apps/preferences/fields.py]: For some reason LDAP is converting the None to a string (which is supposed to happen) as seen [here](https://github.com/tjcsl/ion/blob/master/intranet/apps/preferences/views.py#L97-L102) only for mobile_phone but not any others.

Comment (ezwang) [inline intranet/apps/preferences/fields.py]: I don't think so, I changed the mobile phone field to a phone number, verified it saved, and changed it back to blank, and after a refresh the field displays "None".

Comment (ovkulkarni) [inline intranet/apps/preferences/fields.py]: @ezwang Yes, that's correct, according to the code linked above, it should be set to None.

File: intranet/apps/preferences/fields.py [hunk 2]
```diff
@@ -25,7 +22,5 @@ def to_python(self, value):
         return value
 
     def widget_attrs(self, widget):
```
Comment (tjhsst-ion) [inline intranet/apps/preferences/fields.py]: Codacy Issue found: [Method could be a function](https://www.codacy.com/app/tjcsl/ion/file/2354287857/issues/source?bid=3176160&fileBranchId=3218556#l24)

**Comment by pefoley2** [comment]
Needs a rebase.
Fyi, you can create a temporary branch in the tjcsl/ion repo and then we'll just delete it after this pr is merged.

**Comment by ovkulkarni** [comment]
I'm going to create a new branch as @pefoley2 suggested. Closing PR.

**Pull request closed without merging**